lib/db_utils.go

```diff
@@ -8255,3 +8340,637 @@ func PerformanceBadgerOptions(dir string) badger.Options {
 
 	return opts
 }
+
+// ---------------------------------------------
+// Associations
+// ---------------------------------------------
+
+func DBKeyForUserAssociationByID(associationEntry *UserAssociationEntry) []byte {
+	// AssociationID
+	var key []byte
+	key = append(key, Prefixes.PrefixUserAssociationByID...)
+	key = append(key, associationEntry.AssociationID.ToBytes()...)
+	return key
+}
+
+func DBKeyForUserAssociationByTransactor(associationEntry *UserAssociationEntry) []byte {
+	// TransactorPKID, AssociationType, AssociationValue, TargetUserPKID
+	var key []byte
+	key = append(key, Prefixes.PrefixUserAssociationByTransactor...)
+	key = append(key, associationEntry.TransactorPKID.ToBytes()...)
+	key = append(key, []byte(strings.ToLower(associationEntry.AssociationType))...)
+	key = append(key, []byte{0}...) // Null terminator byte for AssociationType which can vary in length
+	key = append(key, []byte(associationEntry.AssociationValue)...)
+	key = append(key, []byte{0}...) // Null terminator byte for AssociationValue which can vary in length
+	key = append(key, associationEntry.TargetUserPKID.ToBytes()...)
+	key = append(key, associationEntry.AppPKID.ToBytes()...)
+	return key
+}
+
+func DBKeyForUserAssociationByTargetUser(associationEntry *UserAssociationEntry) []byte {
+	// TargetUserPKID, AssociationType, AssociationValue, TransactorPKID
+	var key []byte
+	key = append(key, Prefixes.PrefixUserAssociationByTargetUser...)
+	key = append(key, associationEntry.TargetUserPKID.ToBytes()...)
+	key = append(key, []byte(strings.ToLower(associationEntry.AssociationType))...)
+	key = append(key, []byte{0}...) // Null terminator byte for AssociationType which can vary in length
+	key = append(key, []byte(associationEntry.AssociationValue)...)
+	key = append(key, []byte{0}...) // Null terminator byte for AssociationValue which can vary in length
+	key = append(key, associationEntry.TransactorPKID.ToBytes()...)
+	key = append(key, associationEntry.AppPKID.ToBytes()...)
+	return key
+}
+
+func DBKeyForUserAssociationByUsers(associationEntry *UserAssociationEntry) []byte {
+	// TransactorPKID, TargetUserPKID, AssociationType, AssociationValue
+	var key []byte
+	key = append(key, Prefixes.PrefixUserAssociationByUsers...)
+	key = append(key, associationEntry.TransactorPKID.ToBytes()...)
+	key = append(key, associationEntry.TargetUserPKID.ToBytes()...)
+	key = append(key, []byte(strings.ToLower(associationEntry.AssociationType))...)
+	key = append(key, []byte{0}...) // Null terminator byte for AssociationType which can vary in length
+	key = append(key, []byte(associationEntry.AssociationValue)...)
+	key = append(key, []byte{0}...) // Null terminator byte for AssociationValue which can vary in length
+	key = append(key, associationEntry.AppPKID.ToBytes()...)
+	return key
+}
+
+func DBKeyForPostAssociationByID(associationEntry *PostAssociationEntry) []byte {
```

Comment:
Gonna keep separate for now. Could refactor this PR to use a generic Association (combo of UserAssociations + PostAssociations) but IMO there's some benefits to leaving separate like avoiding generics gotchas and future-proofing in case their use-cases diverge.

Reply:
yeah keeping it separate is best for now.